api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -524,6 +563,91 @@ public boolean equals(Object o) {
     public int hashCode() {
       return Objects.hash(NestedField.class, id, isOptional, name, type);
     }
+
+    private static JsonNode validateDefault(String name, Type type, JsonNode defaultValue) {
+      if (defaultValue != null && !isValidDefault(type, defaultValue)) {
+        throw new ValidationException("Invalid default value for field %s: %s not a %s", name, defaultValue, type);
+      }
+      return defaultValue;
+    }
+
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    private static boolean isValidDefault(Type type, JsonNode defaultValue) {
```

Comment:
Iceberg uses visitors to separate the logic for schema traversal from the logic for individual decisions. Can you please use a visitor?

Reply:
I feel using a `SchemaVisitor` here would be an overkill, since this logic is only validating a `NestField`, it's technically only part of the schema, and even if it's a complex type, we only recursively validate downward, but not upward. Also I feel readability-wise, this recursive implementation is clean enough? it clearly shows the logic is to validate each type of default value case-by-case.

To me, I feel `SchemaVisitor` is mostly used to do holistic schema manipulations/transformations.